deid/utils/actions.py

```diff
@@ -66,10 +67,28 @@ def parse_value(dicom, value, item=None, field=None, funcs=None):
             # item is the lookup, value from the recipe, and field
             # The field is an entire dicom element object
             return item[value_option](dicom=dicom, value=value, field=field, item=item)
-
-        bot.warning("%s is not a valid value type, skipping." % (value_type))
-        return None
-    return value
+        else:
+            bot.warning("%s is not a valid value type, skipping." % (value_type))
+            return None
+
+    # Determine if the value is for an existing field.  If so,
+    # the value must be converted to conform to the appropriate Python type.
+    # Otherwise the field can remain as string and be auto-added as such.
+    if isinstance(field, str) and dicom is not None and field in dicom:
```

Comment:
The pattern I typically do here is have the default be false, and then the block set to true to avoid that extra "else" e.g.,

```python
existingField = False
if case1:
    existingField = True
elif case2:
    existingField = True
```

Reply:
I'm wordy. :)  It's all updated.